service/src/main/java/bio/terra/workspace/service/job/JobBuilder.java

```diff
@@ -119,6 +163,28 @@ private void populateInputParams() {
       addParameter(JobMapKeys.AUTH_USER_INFO.getKeyName(), userRequest);
       addParameter(JobMapKeys.SUBJECT_ID.getKeyName(), userRequest.getSubjectId());
     }
+    if (shouldInsert(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceId)) {
+      addParameter(WorkspaceFlightMapKeys.WORKSPACE_ID, workspaceId);
+    }
+    if (shouldInsert(ResourceKeys.RESOURCE, resource)) {
+      addParameter(ResourceKeys.RESOURCE, resource);
+    }
+    if (shouldInsert(ResourceKeys.RESOURCE_TYPE, resourceType)) {
+      addParameter(ResourceKeys.RESOURCE_TYPE, resourceType);
+    }
+    if (shouldInsert(ResourceKeys.RESOURCE_NAME, resourceName)) {
+      addParameter(ResourceKeys.RESOURCE_NAME, resourceName);
+    }
+    if (shouldInsert(ResourceKeys.STEWARDSHIP_TYPE, stewardshipType)) {
+      addParameter(ResourceKeys.STEWARDSHIP_TYPE, stewardshipType);
+    }
+    if (shouldInsert(ResourceKeys.OPERATION_TYPE, operationType)) {
+      addParameter(ResourceKeys.OPERATION_TYPE, operationType);
+    }
+  }
+
+  private boolean shouldInsert(String mapKey, Object value) {
+    return (value != null && !jobParameterMap.containsKey(mapKey));
```

Comment:
should we handle empty/blank string cases for value?

Reply:
I don't think so. That would mean that someone building a job had deliberately written the input parameter as empty/blank. So we wouldn't want to overwrite that.